app/assets/javascripts/discourse/app/models/login-method.js

```diff
@@ -57,25 +64,22 @@ export default class LoginMethod extends EmberObject {
       return Promise.resolve();
     }
 
-    let authUrl = getURL(`/auth/${this.name}`);
-
     if (reconnect) {
-      params["reconnect"] = true;
+      params.reconnect = true;
     }
 
     if (signup) {
-      params["signup"] = true;
-    }
+      params.signup = true;
 
-    const paramKeys = Object.keys(params);
-    if (paramKeys.length > 0) {
-      authUrl += "?";
-      authUrl += paramKeys
-        .map((k) => `${encodeURIComponent(k)}=${encodeURIComponent(params[k])}`)
-        .join("&");
+      const email = Session.currentProp("email");
```

Comment:
We pick up the email that was stored in the session (if any) and add it to the params when signing up (via external auth).

Reply:
What stores the email in the session? Sorry if it's a simple question, looking at this with fresh eyes and not enough context!